client/dashboard/me/preferences-language/index.tsx

```diff
@@ -0,0 +1,310 @@
+	const selectedLanguage: CalypsoLanguage | undefined = data.language
+		? ( getLanguage( data.language ) as CalypsoLanguage )
+		: undefined;
+
+	const handleSubmit = ( e: React.FormEvent ) => {
+		e.preventDefault();
+		if ( ! localData ) {
+			return;
+		}
+		const mutationData = localData;
+		setLocalData( undefined );
+		setSavingData( mutationData );
+		mutation.mutate( mutationData, {
+			onSuccess: () => {
+				// Ensure the UI picks up the new language by reloading the page.
+				// Add a transient query param so we can show a success notice after reload.
+				const url = new URL( window.location.href );
+				url.searchParams.set( 'settings-saved', '1' );
+				// Replace to avoid adding an extra history entry.
+				window.location.replace( url.toString() );
+			},
+			onError: ( error ) => {
+				// Prepend previous attempted data back into local edits
+				setLocalData( ( current ) => ( { ...mutationData, ...current } ) );
+				createErrorNotice( error.message ?? __( 'Language setting could not be saved.' ), {
+					type: 'snackbar',
+				} );
+			},
+			onSettled: () => {
+				setSavingData( undefined );
+			},
+		} );
+	};
+
+	const isDefaultLanguageSelected = !! data.language && isDefaultLocale( data.language );
+	const showIncompleteLocaleControl =
+		! isDefaultLanguageSelected && !! data.language && isTranslatedIncompletely( data.language );
+	const isSaving = mutation.isPending;
+
+	const isDirty =
+		!! localData &&
+		!! serverData &&
+		Object.entries( localData ).some( ( [ key, value ] ) => {
+			return serverData[ key as keyof UserSettingsPreferences ] !== value;
+		} );
+
+	const hasValidLanguage = !! data?.language && data.language !== '';
+	const canSubmit = ! isSaving && isDirty && hasValidLanguage;
+	const languageForm = {
+		type: 'regular' as const,
+		labelPosition: 'top' as const,
+		fields: [
+			{
+				id: 'interfaceLanguage',
+				label: __( 'Language' ),
+				children: [
+					'language',
+					'use_fallback_for_incomplete_languages',
+					'enable_translator',
+					'i18n_empathy_mode',
+				],
+			},
+		],
+	};
+
+	const languageFields: Field< UserSettingsPreferences >[] = [
+		{
+			id: 'language',
+			label: __( 'Interface language' ),
+			type: 'text',
+			Edit: ( { field, data, onChange } ) => {
+				return (
+					<>
+						<ComboboxControl
+							value={ field.getValue( { item: data } ) ?? '' }
+							label={ __( 'Interface language' ) }
+							onChange={ ( newValue ) => {
+								onChange( {
+									[ field.id ]: newValue,
+								} );
+							} }
+							placeholder={ __( 'Select a language' ) }
+							options={ field.elements || [] }
+							allowReset={ false } // a language is required so we're not allowing to reset it and have an empty state.
+						/>
+
+						<Text variant="muted">
+							{ __(
+								'This is the language of the interface you see across WordPress.com as a whole.'
+							) }
+							{ data && thanksToCommunityTranslator( data ) }
```

Comment:
We can use the `help` prop of the ComboboxControl to display it

Reply:
done in 51c22bae134e0d1fb16514b1a953d3b99cd46996